Stuff from Horfius's fork.

horfius opened the pull request.

horfius committed:
Added the concrete! Needs texturing though.

ScottKillen commented:
Very nice, @horfius. Thank you for the hard work.

It may take a bit for me to get this in as I am still working on some of the basics that need nailing down. But I can merge it. Don't get discouraged if it takes me a bit.

Also, note that as you test and tweak it, you can commit more changes to the same branch and it will update this PR.

Also, A screenshot or two of this in action would be excellent.

ScottKillen commented:
I am also waiting for the Liquid API in forge which Lex is working on.    :hourglass:

horfius committed:
More concrete work

horfius commented:
Added some quick sample textures for the concrete, fixed a few bugs. Still some more work needed, but it's good for now.

horfius committed:
Version checker.

Still needs some work, but it works for the most part. I'd be wary of
some of the methods though.

horfius commented:
The version checker is almost done, but it's 3 am and I'm tired. See you guys soon.

ScottKillen commented:
@horfius I should probably mention that you want to commit that work to a separate branch that is based off of the latest develop/HEAD on the main repo.

If you commit all of that work to the same branch as this work then it will combine both in this one P/R and that would be bad...

See https://guides.github.com/introduction/flow/

ScottKillen commented:
Ah...I see I am tool late. :smile: 

No biggie... Here is what you do:
1. Read the link above
2. Preserve your code somewhere locally.
3. Delete your fork here.
4. Refork.
5. Create and checkout a  branch for concrete.
6. Add your concrete work back in.
7. Commit.
8. Resubmit that PR.
9. Create and checkout a  branch for version checker. This branch should be based on the develop branch.
10. Add your VC work back in.
11. Commit.
12. Submit that PR.

By doing things this way, you can keep the features you add separate and that makes it easier...When I am ready for concrete, I don't have to also have to be ready for VC at the same time.

These rules work for me:
- Never work directly in the develop branch
- Create a new branch based on develop for each feature.

I use [SourceTree](http://www.sourcetreeapp.com/) and that makes all of this _very_ easy. I just wish it had a dark theme. :grinning:

horfius commented:
Okay thanks for letting me know. I haven't used github very much and I don't know how to work it well.

ScottKillen commented:
Good job, @horfius!
